src/lib/Microsoft.Health.Events/EventHubProcessor/ResumableAssignedPartitionProcessor.cs

```diff
@@ -0,0 +1,51 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Events.EventHubProcessor
+{
+    public class ResumableAssignedPartitionProcessor : IResumableEventProcessor
```

Comment:
This service was not something I would have normally created but I think it (or something like it) may be necessary to support both PaaS and OSS codeflows. In PaaS we have a health checker that stops and restarts an IResumableEventProcessor when the health checks succeed or fail. Even though AssignedPartitionProcessor could itself be a IResumableEventProcessor we don't really want to stop and restart that in PaaS because the AssignedPartitionProcessor needs some information in order to run and it gets that information from the outer loop (PartitionLockingBackgroundService) which is responsible for registering the processor and also claiming the partitions that the AssignedPartitionProcessor should read from.

I agree that it is weird that a background service wrapping another background service. If I were just coding for OSS then we only need PartitionLockingBackgroundService and AssignedPartitionProcessor but to support the health checker logic we have in PaaS we kind of need another abstraction that implements IResumableEventProcessor which is what I did here with ResumableAssignedPartitionProcessor, or we need a bit of a refactor. I am open to refactoring more but wanted to explain the thought process.

Reply:
added a flowchart: https://github.com/microsoft/iomt-fhir/blob/personal/wiyochum/partition-locking-prototype/docs/PartitionLocking.md

Reply:
I guess, my question then is what is the need for the PartitionLockingBackgroundService to be a background service?  Originally I was concerned we had two places where the PartitionLockingBackgroundService is started/controlled.  One in the resumable processor and one as a background service.  Upon further investigation I didn't find any place where we were invoking AddHostedService in the DI on the PartitionLockingBackgroundService which means it isn't running as a background service in the processor.  If we are just using the background service contract, I think we should create our own interface for the control flow and not reuse the background service construct since it can cause confusion.